### ISSUE TITLE
Documents that don't contain an _id should have one generated

### ISSUE THREAD

**Issue by bfink13** [bwaldvogel/mongo-java-server#76]
Normally drivers are responsible for generating the _id field on inserted documents. However, it appears that certain usages of codecs will cause the _id to not be generated. In those cases, we should generate an ObjectID and attach it to the document prior to inserting. This also matches the behavior of using the Mongo CLI directly:
```
rs0:PRIMARY> db.test2.insert({"blah":"blah"});
WriteResult({ "nInserted" : 1 })
rs0:PRIMARY> db.test2.find()
{ "_id" : ObjectId("5d163def5a8bb6a0cc777e65"), "blah" : "blah" }
rs0:PRIMARY> 
```

**Comment by bwaldvogel** [comment]
Implemented in b344260

**Comment by bwaldvogel** [comment]
The fix is part of version `1.18.0`.
